components/gitpod-cli/cmd/rebuild.go

```diff
@@ -241,24 +241,35 @@ func runRebuild(ctx context.Context, supervisorClient *supervisor.SupervisorClie
 		return err
 	}
 
-	// we don't pass context on purporse to handle graceful shutdown
-	// and output all logs properly below
-	runCmd := exec.Command(
-		dockerPath,
+	var runArgs []string
+	runArgs = append(runArgs,
 		"run",
 		"--rm",
 		"--user", "root",
 		"--privileged",
 		"--label", "gp-rebuild=true",
 		"--env-file", envFile,
-
-		// ports
-		"-p", "24999:22999", // supervisor
-		"-p", "25000:23000", // Web IDE
-		"-p", "25001:23001", // SSH
-		// 23002 dekstop IDE port, but it is covered by debug workspace proxy
-		"-p", "25003:23003", // debug workspace proxy
-
+	)
+	// ports
+	if !rebuildOpts.HostNetwork {
+		runArgs = append(runArgs,
+			"-p", "24999:22999", // supervisor api
+			"-p", "25000:23000", // Web IDE
+			"-p", "25001:23001", // SSH
+			// 23002 dekstop IDE port, but it is covered by debug workspace proxy
+			"-p", "25003:23003", // debug workspace proxy
+		)
+	} else {
+		runArgs = append(runArgs,
```

Comment:
rebase on Pudong's PR to revert it, i.e. when we can run docker isolated in the debug workspace